File: extension-system/sdk-node/src/decision.ts
import type {
  DecisionAskParams,
  DecisionAskResult,
} from "@mocu/extension-contracts";

import { DECISION_ASK_METHOD } from "@mocu/extension-contracts";

import { JsonRpcProtocolClient } from "./protocol-client.js";

/**
 * Lets an extension ask typed probabilistic questions through Mocu's
 * configured Jev decision model (OpenRouter Decisions API).
 */
export class DecisionApi {
  constructor(private readonly client: JsonRpcProtocolClient) {}

  async ask(
    params: DecisionAskParams,
  ): Promise<DecisionAskResult> {
    if (!params) {
      throw new Error("Decision ask requires params.");
    }

    if (params.state === undefined || params.state === null) {
      throw new Error("Decision ask requires a 'state' to reason about.");
    }

    if (
      !params.questions ||
      typeof params.questions !== "object" ||
      Array.isArray(params.questions) ||
      Object.keys(params.questions).length === 0
    ) {
      throw new Error(
        "Decision ask requires a non-empty 'questions' object.",
      );
    }

    return this.client.request<DecisionAskResult>(
      DECISION_ASK_METHOD,
      {
        state: params.state,
        questions: params.questions,
      },
    );
  }
}

export default DecisionApi;

File: extension-system/sdk-python/mocu_extension_sdk/decision.py
from __future__ import annotations

from typing import Any

from .protocol import JsonRpcProtocol


class DecisionApi:
    """
    Lets an extension ask typed probabilistic questions through Mocu's
    configured Jev decision model (OpenRouter Decisions API). The model does
    not generate text; it returns calibrated probabilities.
    """

    def __init__(self, protocol: JsonRpcProtocol) -> None:
        self._protocol = protocol

    def ask(
        self,
        *,
        state: Any,
        questions: dict[str, Any],
    ) -> Any:
        if state is None:
            raise ValueError("Decision ask requires a 'state' to reason about.")

        if (
            not isinstance(questions, dict)
            or not questions
        ):
            raise ValueError(
                "Decision ask requires a non-empty 'questions' object."
            )

        return self._protocol.request(
            "mocu.decision.ask",
            {
                "state": state,
                "questions": questions,
            },
        )
